include/libsemigroups/bmat-adapters.hpp

```diff
@@ -28,19 +32,57 @@
 #include "exception.hpp"  // for LIBSEMIGROUPS_EXCEPTION
 #include "matrix.hpp"     // for BMat
 
+#include "libsemigroups/detail/containers.hpp"  // for StaticVector1
+
 namespace libsemigroups {
+  //! \defgroup adapters_bmat_group Adapters for BMat
+  //!
+  //! \ingroup adapters_group
+  //!
+  //! \brief Documentation for the specialization of adapters that allow
+  //! ``BMat`` types to be used in the algorithms of `libsemigroups`.
+  //!
+  //! This page contains the documentation of the functionality in
+  //! `libsemigroups` that adapts ``BMat`` types for use with the
+  //! algorithms in libsemigroups.
+  //!
+  //! @{
   ////////////////////////////////////////////////////////////////////////
   // ImageRight/LeftAction - BMat
   ////////////////////////////////////////////////////////////////////////
 
+  //! \brief Specialisation of the ImageRightAction adapter for ``BMat`` and
+  //! containers of ``BitSet``.
+  //!
+  //! Defined in ``bmat-adapters.hpp``.
+  //!
+  //! Specialization of the ImageRightAction adapter for ``BMat`` and containers
+  //! of ``BitSet``.
+  //!
+  //! \sa ImageRightAction.
+  //!
+  //! \warning
+  //! The template type `Mat` must be a `BMat` type (``IsBMat<Mat>`` must be
+  //! true), and the ``value_type`` of the template type ``Container`` must be a
+  //! bit set (``IsBitSet<typename Container::value_type>>`` must be true). If
+  //! not, template substitution will fail.
   // T = StaticVector1<BitSet<N>, N> or std::vector<BitSet<N>>
   // Possibly further container when value_type is BitSet.
   template <typename Container, typename Mat>
+#ifndef PARSED_BY_DOXYGEN
   struct ImageRightAction<
       Mat,
       Container,
-      std::enable_if_t<IsBMat<Mat>
-                       && IsBitSet<typename Container::value_type>>> {
+      std::enable_if_t<IsBMat<Mat> && IsBitSet<typename Container::value_type>>>
+#else
+  struct ImageRightAction<Mat, Container>
+#endif
+  {
+    // TODO(now) Are pt and x the right way round in the doc?
```

Comment:
@james-d-mitchell I'm not confident that I have got `x` and `pt` the right way round here. What do you think?